src/filesystem.cpp

```diff
@@ -1776,4 +1776,23 @@ std::string get_localized_path(const std::string& file, const std::string& suff)
 	return "";
 }
 
+std::string get_addon_id_from_path(const std::string& location)
+{
+	bfs::path path = bfs::absolute(location);
+	bool use_next = false;
+	std::string addon_base = filesystem::base_name(filesystem::get_addons_dir());
```

Comment:
Why does this not compare the absolute paths and if the addons dir is a prefix then pick the first dir after the prefix? Looking for an "add-ons" dir with no context seems like an unnecessary approximation.

Reply:
`bfs::path` isn't indexed as far as I know, so there doesn't appear to be a way to say "get the next directory in the path after X".

Reply:
Updated.